src/sql/platform/connection/test/common/providerConnectionInfo.test.ts

```diff
@@ -216,10 +216,11 @@ suite('SQL ProviderConnectionInfo tests', () => {
 
 	test('getOptionsKey should create a valid unique id', () => {
 		let options: { [key: string]: string } = {};
+		// Setting custom options are not yet considered for profile identity
```

Comment:
It would be better to make a separate test for this specific functionality - `getOptionsKey should generate the same ID regardless of custom option values`

Reply:
That's not a use-case, it's a bug IMO.

Reply:
Then you should remove this check completely if that's the case.

But I'm not sure I quality this as a bug currently - it's very much intended behavior. As discussed though, there is potential for us changing that behavior, but that'd be an enhancement/functionality change at that point.